src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,726 @@
+
+    auto it = blocks.begin();
+
+    uint64_t block_count = 0;
+    while (it != blocks.end())
+    {
+      cryptonote::rpc::block_with_transactions& bwt = res.blocks[block_count];
+
+      block& blk = it->first;
+      bwt.block = blk;
+
+      std::list<transaction>& txs = it->second;
+
+      cryptonote::rpc::block_output_indices& indices = res.output_indices[block_count];
+
+      // assume each block returned is returned with all its transactions
+      // in the correct order.
+      auto tx_it = txs.begin();
+      for (crypto::hash& h : blk.tx_hashes)
+      {
+        bwt.transactions.emplace(h, *tx_it);
+        tx_it++;
+
+        cryptonote::rpc::tx_output_indices tx_indices;
+        bool r = m_core.get_tx_outputs_gindexs(h, tx_indices);
+        if (!r)
+        {
+          res.status = Message::STATUS_FAILED;
+          res.error_details = "core::get_tx_outputs_gindexs() returned false";
+          return;
+        }
+
+        indices.push_back(tx_indices);
+      }
+
+      it++;
+      block_count++;
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(GetHashesFast::Request& req, GetHashesFast::Response& res)
+  {
+    res.start_height = req.start_height;
+
+    auto& chain = m_core.get_blockchain_storage();
+
+    if (!chain.find_blockchain_supplement(req.known_hashes, res.hashes, res.start_height, res.current_height))
+    {
+      res.status = Message::STATUS_FAILED;
+      res.error_details = "Blockchain::find_blockchain_supplement() returned false";
+      return;
+    }
+
+    res.status = Message::STATUS_OK;
+  }
+
+  void DaemonHandler::handle(GetTransactions::Request& req, GetTransactions::Response& res)
+  {
+    std::list<cryptonote::transaction> found_txs;
+    std::list<crypto::hash> missed_hashes;
+
+    bool r = m_core.get_transactions(req.tx_hashes, found_txs, missed_hashes);
+
+    // TODO: handle return false (std::exception caught above)
+    if (!r)
+    {
+    }
+
+    uint64_t num_found = found_txs.size();
+
+    // std::list is annoying
+    std::vector<cryptonote::transaction> found_txs_vec
+    {
+      std::make_move_iterator(std::begin(found_txs)),
+      std::make_move_iterator(std::end(found_txs))
+    };
+
+    std::vector<crypto::hash> missed_vec
+    {
+      std::make_move_iterator(std::begin(missed_hashes)),
+      std::make_move_iterator(std::end(missed_hashes))
+    };
+
+    std::vector<uint64_t> heights(num_found);
+    std::vector<bool> in_pool(num_found, false);
+    std::vector<crypto::hash> found_hashes(num_found);
+
+    for (size_t i=0; i < num_found; i++)
+    {
+      found_hashes[i] = get_transaction_hash(found_txs_vec[i]);
+      heights[i] = m_core.get_blockchain_storage().get_db().get_tx_block_height(found_hashes[i]);
+    }
+
+    // if any missing from blockchain, check in tx pool
+    if (!missed_vec.empty())
+    {
+      std::list<cryptonote::transaction> pool_txs;
+
+      m_core.get_pool_transactions(pool_txs);
```

Comment:
It literally always returns true.  As for *that*, there might be something there to fix...